### ISSUE TITLE
DAY 7 — PROGRESS

### ISSUE THREAD

**Issue by charisioskoutsos-sketch** [charisioskoutsos-sketch/Isef-project-#7]
DATE: September 20, 2026
DAY: 7

OBJECTIVE

To understand the fundamental structure and training process of neural networks.

WORK COMPLETED

✓ Neural networks
✓ Artificial neurons
✓ Weights
✓ Biases
✓ Activation functions
✓ ReLU
✓ Sigmoid
✓ Neural-network layers
✓ Forward pass
✓ Backpropagation
✓ Gradients
✓ Connection between gradients and optimization
✓ Introduction to CNNs
✓ Connection between CNNs and solar magnetograms

UNDERSTANDING

✓ Understood how an artificial neuron transforms input values into an output.
✓ Understood the roles of weights and biases.
✓ Understood why activation functions are needed in neural networks.
✓ Understood the roles of input, hidden, and output layers.
✓ Understood the basic forward-pass and backpropagation process.
✓ Understood how gradients are used to update model parameters.
✓ Understood why CNNs are relevant for extracting spatial information from magnetograms.

CURRENT STATUS

Day 7 theory completed. Exercise questions remain to be answered and reviewed.

NEXT STEP

→ Review Day 7 exercises
→ Convolutional Neural Networks
→ Convolution, filters, feature maps, and pooling
